app/services/imports/case_logs_import_service.rb

```diff
@@ -0,0 +1,278 @@
+    #       attributes["needstype"] == GN_SH[:general_needs] &&
+    #       landlord == PRP_LA[:private_registered_provider]
+    #     1
+    #   elsif attributes["renttype"] == SR_AR_IR[:social_rent] &&
+    #       attributes["needstype"] == GN_SH[:supported_housing] &&
+    #       landlord == PRP_LA[:private_registered_provider]
+    #     2
+    #   elsif attributes["renttype"] == SR_AR_IR[:social_rent] &&
+    #       attributes["needstype"] == GN_SH[:general_needs] &&
+    #       landlord == PRP_LA[:local_authority]
+    #     3
+    #   elsif attributes["renttype"] == SR_AR_IR[:social_rent] &&
+    #       attributes["needstype"] == GN_SH[:supported_housing] &&
+    #       landlord == PRP_LA[:local_authority]
+    #     4
+    #   elsif attributes["renttype"] == SR_AR_IR[:affordable_rent] &&
+    #       attributes["needstype"] == GN_SH[:general_needs] &&
+    #       landlord == PRP_LA[:private_registered_provider]
+    #     5
+    #   elsif attributes["renttype"] == SR_AR_IR[:affordable_rent] &&
+    #       attributes["needstype"] == GN_SH[:supported_housing] &&
+    #       landlord == PRP_LA[:private_registered_provider]
+    #     6
+    #   elsif attributes["renttype"] == SR_AR_IR[:affordable_rent] &&
+    #       attributes["needstype"] == GN_SH[:general_needs] &&
+    #       landlord == PRP_LA[:local_authority]
+    #     7
+    #   elsif attributes["renttype"] == SR_AR_IR[:affordable_rent] &&
+    #       attributes["needstype"] == GN_SH[:supported_housing] &&
+    #       landlord == PRP_LA[:local_authority]
+    #     8
+    #   elsif attributes["renttype"] == SR_AR_IR[:intermediate_rent] &&
+    #       attributes["needstype"] == GN_SH[:general_needs] &&
+    #       landlord == PRP_LA[:private_registered_provider]
+    #     9
+    #   elsif attributes["renttype"] == SR_AR_IR[:intermediate_rent] &&
+    #       attributes["needstype"] == GN_SH[:supported_housing] &&
+    #       landlord == PRP_LA[:private_registered_provider]
+    #     10
+    #   elsif attributes["renttype"] == SR_AR_IR[:intermediate_rent] &&
+    #       attributes["needstype"] == GN_SH[:general_needs] &&
+    #       landlord == PRP_LA[:local_authority]
+    #     11
+    #   elsif attributes["renttype"] == SR_AR_IR[:intermediate_rent] &&
+    #       attributes["needstype"] == GN_SH[:supported_housing] &&
+    #       landlord == PRP_LA[:local_authority]
+    #     12
+    #   else
+    #     raise "Could not infer rent type with rentype:#{attributes['renttype']} / needstype:#{attributes['needstype']} / landlord:#{landlord}"
+    #   end
+    # end
+
+    def find_organisation_id(xml_doc, field)
+      old_visible_id = field_value(xml_doc, "xmlns", field).to_i
+      landlord = field_value(xml_doc, "xmlns", "Landlord").to_i
+
+      organisation = Organisation.find_by(old_visible_id:)
+      # Quick hack: should be removed when all organisations are imported
+      # Will fail in the future if the organisation is missing
```

Comment:
Might be better (after the first organisation is successfully imported perhaps), to have a single "import org" task that runs the individual tasks in order for an org such that you know an org exists by the time you're here and don't need to handle this case?